src/app/modules/form/admin-form/admin-form.middlewares.ts

```diff
@@ -82,3 +64,49 @@ export const getWebhookSettingsValidator = celebrate({
     userEmail: Joi.string().email().optional(),
   }),
 })
+
+/**
+ * Joi validator for POST /forms/:formId/workflow/ route.
+ */
+export const createWorkflowStepValidator = celebrate({
+  [Segments.BODY]: Joi.object({
+    workflow_type: Joi.string().valid(...Object.values(WorkflowType)),
+    emails: Joi.when('workflow_type', {
+      is: WorkflowType.Static,
+      then: Joi.array().items(Joi.string().email()).required(),
+    }),
+    field: Joi.when('workflow_type', {
+      is: WorkflowType.Dynamic,
+      then: Joi.string().hex().length(24).required(),
```

Comment:
I understand that this is to ensure that this fits the our current ObjectId restriction. However, I think these are implementation details that the application doesn't have to strictly check for. 

i.e., 
whether mongodb is using a 24-char hex string or 76-char base64 or whether our application restricts to only allow generation of 24-char hex strings.

Reply:
i added it as a means to harden our request validation security by reducing the number of strings that would pass our validation. 

though if preferred, i can remove this as well which may have the benefit of having less things to update if we change from using `hex().length(24)` to something else